assets/extensions/all-extensions.js

```diff
@@ -1,73 +1,161 @@
 /**
- * WordPress dependencies
+ * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { keyBy } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import Card from './card';
+import { Grid, Col } from './grid';
 
-const AllExtensions = ( { extensions } ) => {
-	const featuredExtensions = extensions.filter(
-		( extension ) => extension.is_featured
-	);
-
-	return (
-		<>
-			{ featuredExtensions.length > 0 && (
-				<section className="sensei-extensions__section sensei-extensions__grid__col --col-12">
-					<h2 className="sensei-extensions__section__title">
-						{ __( 'Featured', 'sensei-lms' ) }
-					</h2>
-					<ul className="sensei-extensions__section__content sensei-extensions__featured-list">
-						{ featuredExtensions.map( ( extension ) => (
-							<li
-								key={ extension.product_slug }
-								className="sensei-extensions__featured-list__item"
-							>
-								<div className="sensei-extensions__featured-list__card-wrapper">
-									<Card extension={ extension } />
-								</div>
-							</li>
-						) ) }
-					</ul>
-				</section>
-			) }
+// TODO: Get from API.
+const extensionsSkeleton = [
+	{
+		id: 'featured',
+		columns: 12,
+		type: 'featured-list',
+		title: 'Featured',
+		itemSlugs: [
+			'sensei-wc-paid-courses',
+			'sensei-content-drip',
+			'sensei-certificates',
+		],
```

Comment:
I think this could be more generic, with extension cards just one kind of item — that would allow us to put other things in there, like a message or a theme promo. For maximum flexibility we could have a HTML card type that can output whatever the server sends. 

Also, for the extension cards, would it make sense to add an `options` or `props` attribute that can be passed to the component? It could allow greater control over possible variations of an extension card, like adding extra classes to a certain item. 

```
items: [ {
  type: 'extension-card',
  slug: 'sensei-content-drip',
  props: {
    comingSoon: true,
    className: 'featured-purple'
    },
  },
]
```

Reply:
Examples for other content in WooCommerce's extensions page:
<img width="1217" alt="image" src="https://user-images.githubusercontent.com/176949/115895792-18cd0280-a45b-11eb-85f3-6a8e76c6011d.png">

Reply:
Nice idea about the props customization. Added it here: https://github.com/Automattic/sensei/pull/4197/commits/adbf4848fbb69591654ea6ae7103882422ca972a

I was planning to implement the custom card in a separate PR after we decided if we go ahead, but I took the opportunity to already implement it: https://github.com/Automattic/sensei/pull/4197/commits/511ca605ab325c669a89a7572782bc1c91c69231